src/components/Dropdown/Dropdown.tsx

```diff
@@ -689,7 +696,12 @@ export default class Dropdown extends AutoControlledComponent<
     }
 
     // we don't have event for it, but want to keep the event handling interface, event is empty.
-    _.invoke(this.props, 'onSelectedChange', {}, { ...this.props, value: newValue })
+    _.invoke(
+      this.props,
+      'onSelectedChange',
+      {},
+      { ...this.props, searchQuery: '', value: newValue },
```

Comment:
Previously we made a reset in `setState()` of `searchQuery`, but this change was not passed in `onSelectedChange`